cmd/crane/cmd/serve.go

```diff
@@ -0,0 +1,69 @@
+// Copyright 2023 Google LLC All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"errors"
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/spf13/cobra"
+
+	"github.com/google/go-containerregistry/pkg/registry"
+)
+
+func NewCmdServe() *cobra.Command {
+	port := os.Getenv("PORT")
+	if port == "" {
```

Comment:
could we support `kernel selected port (0)`?

Reply:
that'd be really helpful for spawning multiple instances of registry. my use case is bazel actions

Reply:
Sure, that seems reasonable. Is there a way you'd prefer to do that? Some quick googling found [`freeport`](https://pkg.go.dev/github.com/phayes/freeport#GetFreePort) which looks promising, but I've never used it.

Reply:
my implementation of registry looks like this; https://github.com/bazel-contrib/rules_oci/blob/main/e2e/custom_registry/registry/registry.go using `net.Listen`

Reply:
I prefer this as it guarantees that you'll be allocated a port by the kernel.